modules/renter/download_test.go

```diff
@@ -20,6 +23,8 @@ type testHost struct {
 	// used to simulate real-world conditions
 	delay    time.Duration // transfers will take this long
 	failRate int           // transfers will randomly fail with probability 1/failRate
+
+	sync.Mutex
```

Comment:
yeah. In this case, this issue is that `repair` (and `upload`, for that matter) call the methods of `uploader` concurrently. I don't see an easy way to get around that, but it should probably be noted in the docstring of the `uploader` interface.